lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart

```diff
@@ -2,47 +2,42 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
 import 'package:ui/src/engine.dart';
 import 'package:ui/ui.dart' as ui;
 
 /// Tracks the [FlutterView]s focus changes.
 final class ViewFocusBinding {
   /// Creates a [ViewFocusBinding] instance.
-  ViewFocusBinding._();
+  ViewFocusBinding(this._viewManager) {
+    _init();
+  }
+
+  final FlutterViewManager _viewManager;
+
+  Stream<ui.ViewFocusEvent> get onViewFocusChange => _onViewFocusChangeController.stream;
+  final StreamController<ui.ViewFocusEvent> _onViewFocusChangeController =
+      StreamController<ui.ViewFocusEvent>.broadcast(sync: true);
```

Comment:
`sync: true` may have exciting consequences, according to the docs :P

Reply:
Also: you can use `onListen` and `onCancel` on the `StreamController` to know if anybody is listening to the stream. You could use those events to call the code that you have in `_init` as late as needed and in `dispose` as early as possible.

The actual implementation of `dispose` could be: `_onViewFocusChangeController.close()`? (which should cause all the attached streams to cancel, if any?)

Reply:
> sync: true may have exciting consequences, according to the docs :P

Great call removed.

> Also: you can use onListen and onCancel on the StreamController to know if anybody is listening to the stream. You could use those events to call the code that you have in _init as late as needed and in dispose as early as possible.

I still prefer the existing implementation (realized that the early we listen for these focus changes, the better). 
Are there any flaws with it? Happy to change if you feel strongly about it.

Reply:
We normally use callbacks rather than streams, and callbacks are synchronous. So I wouldn't say that using a synchronous stream would be non-idiomatic. In fact, I'd say that using streams in the first place would be non-idiomatic, whether they are sync or not :)

I think this depends on what semantics we actually want here. Sync vs async will dictate the timing, potentially race conditions.

Reply:
Ah yes, the problem with tests is that we're not waiting for the event to propagate and have the listener do its thing. `sync` works in this case because it makes the callback sync, and also is valid because it's just propagating the async-ness of a browser event.

This could be made fully async. I suspect tests would be fixed just by adding a listener on the same stream, awaiting for the next event to happen (`await blah.stream.first`), and THEN checking the DOM right after :)

(We just need that signal to know that the DOM has been modified)

Reply:
Having said this, it's true that the callback is simple, and there's no clear win on why this should emit a stream, so YAGNI says: do the callback.

Apologies for the churn!

Reply:
Nothing to apologize for @ditman ! Getting the right design abstractions is hard and I'm glad we chatted about different options and its trade-offs!